### PR TITLE
Update build process

### PR DESCRIPTION
**Pull request by alwaysblank**
This updates `gulp` and all the `gulp-*` dependencies to the latest version(s) and, most notably, removes `node-sass` so we don't have to deal with an interminable install process. Gulp 4 changes the way gulpfiles work a little big, hence the changes here. The build process should also be a little (~2sec) faster.

This also changes the build script to use [`npm ci`](https://docs.npmjs.com/cli/v7/commands/npm-ci) instead of `npm install` , which should mean installs run a bit faster in automated environments (i.e. GitHub Actions).

It does _not_ touch the webpack config because this is apparently running on Webpack 1, and I didn't want to touch that just yet.

_I don't have a way to test if this is building the correct things._ It builds without errors and there are files in the dist folder that certain have reasonable content, but someone with a full environment should check this to make sure things actually look at work correctly.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v14.17.0
```
Comment (kadamwhite) [inline .nvmrc:1]: Let's pin v12 for now, since that's what's in use on the Altis build environment.

**Review by kadamwhite — CHANGES_REQUESTED** [review]
Looks good! I'll try to spin up a test environment, or we can yolo it :grin: For now, the only thing I'd request to be changed is the pinned Node version.

:tada: Hooray for another repo off of libsass

File: .nvmrc (hunk 1 above)
Comment (alwaysblank) [reply .nvmrc]: 813d3c3 changes this to use node 12, and it _seems_ to work on my machine, but `npm install` throws some errors that seem to be about node-gyp, so I'm not sure how it'll behave in other environments. [node 12 is technically unsupported](https://github.com/dlmanning/gulp-sass#support) by `gulp-sass` so if this fails in a different env we may need to reconsider how to set up this build.

Is updating Altis to 14 (or 16 which iirc is the next LTS) on the roadmap?

**Comment by kadamwhite** [comment]
@alwaysblank I've made a few subsequent adjustments based on things I've observed in some of the other projects:

- Install nvm and set v12 in the `.build-script`
- Wrap the `gulp` command in a `package.json` script to obviate specifying the executable path
- Loosen nvmrc range to permit any v12, to simplify things

Let me know if you think these look good, and we can yolo it on out for testing :grin:

**Review by kadamwhite — APPROVED** [review]

**Pull request merged**